apps/charterafrica/src/lib/openAfrica.js

```diff
@@ -1,84 +1,131 @@
+import allCountries from "@/charterafrica/lib/data/json/countries";
 import fetchJson from "@/charterafrica/utils/fetchJson";
 
 const BASE_DOCUMENTS_URL = "https://openafrica.net/api/3/action/";
 
-async function formatDatasets(data) {
-  const {
-    result: {
-      count,
-      facets: { tags: allTags, groups: allCountries },
-      results,
-    },
-  } = data || {};
+const PAGE_SIZE = 10;
 
-  const formattedDatasets = results?.map((dataset) => {
+async function packageSearch(params) {
+  try {
+    const response = await fetchJson.get(
+      `${BASE_DOCUMENTS_URL}package_search`,
+      {
+        params,
+      }
+    );
+    return response;
+  } catch (error) {
+    return error;
+  }
+}
+
+function formatResources(resources, author) {
+  return resources?.map((resource) => {
+    const {
+      created,
+      id,
+      last_modified: updated,
+      url,
+      format,
+      name,
+      description,
+    } = resource;
+    return {
+      author,
+      created,
+      description,
+      format,
+      id,
+      name,
+      updated,
+      url,
+    };
+  });
+}
+
+function formatDatasets(datasets) {
+  return datasets?.map((dataset) => {
     const {
       author,
+      id,
       metadata_created: created,
       metadata_modified: updated,
       name,
       notes,
       resources,
       title,
       type,
+      url,
     } = dataset;
 
-    const formattedResources = resources?.map((resource) => {
-      const { url, format, name: resourceName, description } = resource;
-      return {
-        description,
-        format,
-        name: resourceName,
-        url,
-      };
-    });
+    const formattedResources = formatResources(resources, author);
 
-    const allDocumentFormats = [
-      ...new Set(formattedResources?.map((resource) => resource.format)),
-    ];
+    const formatsSet = new Set(resources?.map((r) => r.format));
+    const formats = Array.from(formatsSet);
 
     return {
       author,
       created,
-      documents: formattedResources,
-      formats: allDocumentFormats,
+      formats,
+      formattedResources,
+      id,
       name,
       notes,
+      url: url?.trim(),
       title,
       type,
       updated,
     };
   });
+}
+
+function formatResponse(data) {
+  const { result: { count, facets: { tags, groups }, results } = {} } =
+    data || {};
+
+  const datasets = formatDatasets(results);
+  const sortStrings = (a, b) => a.localeCompare(b);
+  const uniqueCountries = Object.keys(groups || {}).sort(sortStrings);
+  const validCountries = uniqueCountries.filter((country) =>
+    allCountries.africa
+      .map((c) => c.toLowerCase())
```

Comment:
We have to lower-case **everytime**? Why not just convert them to lower case **once** and store them in allCountries?